include/mxnet/ndarray.h

```diff
@@ -831,7 +855,7 @@ class NDArray {
 
     Chunk(const NDArrayStorageType storage_type_, const TBlob &data,
           const std::vector<TBlob> &aux_data, int dev_id)
-        : static_data(true), delay_alloc(false), storage_type(storage_type_) {
+        : static_data(true), delay_alloc(false), storage_type(storage_type_), deleter_(nullptr) {
```

Comment:
We cannot plug deleter to the TBlob and Chunk, this seems will cause more problems than what it can solve.

std::shared_ptr<Chunk> do support customized deleter already, and maybe that can be used to help solve this issue of having to hide a customized deleter for NDArray